### DISCUSSION TITLE
Cannot use GPU, CuPy is not installed

### DISCUSSION THREAD

**Discussion by Saswati-Project** [explosion/spaCy#13567] [Help: Installation] [answered]
I have a GPU enabled machine. 

O.S: Ubuntu 20.04.6 LTS
nvcc version: 12.2
Nvidia Driver Version: 535.183.01
Pytorch version 2.3.1+cu121
spaCy version    3.7.5
Python version   3.8.10
Pipelines :       en_core_web_sm (3.7.1)

I am using a virtual environment. 

I received the output **True** for the following command

```
>>>import torch
>>> torch.cuda.is_available()
```
However, when I ran 

```
>>>import spacy
>>> spacy.require_gpu()
```

I got the error: 

```
File "<stdin>", line 1, in <module>
  File "/home/abc/spacy_test_env/lib/python3.8/site-packages/thinc/util.py", line 230, in require_gpu
    raise ValueError("Cannot use GPU, CuPy is not installed")
ValueError: Cannot use GPU, CuPy is not installed
```
I installed Spacy with GPU enabled version with the command **pip install -U 'spacy[cuda122]'**
While installing with the above command I got an warning like **spacy 3.7.5 does not provide the extra 'cuda122'**

Can anyone please tell me what is the error in my installation?

**Comment by surfingkaka** [comment, marked as answer]
Try installing `pip install cupy` 
this worked for me

**Reply by Saswati-Project** [reply]
Hi @surfingkaka . Thank you for the suggestion. I did that but the error is still there.

**Reply by surfingkaka** [reply]
try this:

Pip install cupy-wheel

**Reply by Saswati-Project** [reply]
Error is still there. No change after Pip install cupy-wheel

**Reply by Saswati-Project** [reply]
I again run the command pip install cupy and now it is working. Thank you.
